crates/ruff/src/checkers/noqa.rs

```diff
@@ -242,11 +247,20 @@ pub fn check_noqa(
                                         Location::new(row + 1, start - leading_spaces),
                                         Location::new(row + 1, end + trailing_spaces),
                                     ));
-                                } else {
+                                } else if lines[row]
+                                    .chars()
+                                    .nth(end + 1)
+                                    .map_or(false, |c| c == '#')
+                                {
                                     diagnostic.amend(Fix::deletion(
                                         Location::new(row + 1, start),
                                         Location::new(row + 1, end + trailing_spaces),
                                     ));
+                                } else {
+                                    diagnostic.amend(Fix::deletion(
+                                        Location::new(row + 1, start + 1),
+                                        Location::new(row + 1, end),
+                                    ));
```

Comment:
Nit: Is this section a 1:1 copy of the above? Can we extract it in a private-function